main.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+import argparse
+from src.gcms_metadata_generator import GCMSMetabolomicsMetadataGenerator
+from src.lcms_metadata_generator import LCMSLipidomicsMetadataGenerator
+from src.nom_metadata_generator import NOMMetadataGenerator
+
+
+def main():
+    """
+    Parse command-line arguments and run the a metadata generator.
+
+    This function sets up argument parsing for the script, initializes a
+    the chosen metadata generator instance with the provided arguments, and
+    runs the metadata generation process.
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+
+    Side Effects
+    ------------
+    Generates a JSON file with the database dump at the specified path.
+
+    Command-line Arguments
+    ----------------------
+    --generator : str
+        The name of the metadata generator to use.
+        Options:
+        lcms for 'LipidomicsMetadataGenerator'
+        gcms 'GCMSMetabolomicsMetadataGenerator'
+        nom for 'NOMMetadataGenerator'.
+    --metadata_file : str
+        Path to the input CSV metadata file.
+        Example: See example_metadata_file.csv in this directory.
+    --database_dump_json_path : str
+        Path where the output database dump JSON file will be saved.
+    --raw_data_url : str
+        URL base for the raw data files.
+        Example: 'https://nmdcdemo.emsl.pnnl.gov/nom/1000soils/raw/'
+    --process_data_url : str
+        URL base for the processed data files.
+        Example: 'https://nmdcdemo.emsl.pnnl.gov/nom/1000soils/results/'
+    --minting_config_creds : str
```

Comment:
Is this optional?  Can we update the documentation to note that it's optional and will try to use env variables if not declared?